### ISSUE TITLE
Incorrect comment highlighting with recent update

### ISSUE THREAD

**Issue by kankaristo** [Dart-Code/Dart-Code#1573]
I use the following method to quickly and easily comment out code blocks:
```c
/*/
code();
// */
```

Adding a single `*` on the first line uncomments the code block, so you can toggle the code block with a single character:
```c
/**/
code();
// */
```

You can also toggle between 2 different code blocks with a single character, if you add `*` to the first line here:
```c
/*/
code("A");
/*/
code("B");
// */
```

-----

The same works in all languages that use `/* */` for multiline comments (the C "family", JavaScript, JSON with comments, and a similar thing even works in Lua). Dart is a bit weird in that it **requires** the space in `// */` or the code will fail to compile, but that's a "bug" in Dart, not in this package.

-----

However, a recent change in this package (maybe related to https://github.com/Dart-Code/Dart-Code/issues/1498) breaks syntax highlighting.

The last line (`// */`) seems to be treated as a one line comment, and the end of the multiline comment is ignored, so the rest of the file is highlighted as a comment:
![Screenshot from 2019-04-02 10-54-19](https://user-images.githubusercontent.com/3522333/55385398-b7652e80-5535-11e9-8ac5-3294836c3200.png)

 If there's a later `*/` (not part of another comment), the comment highlighting stops there, but it's (correctly) marked as a syntax error:
![Screenshot from 2019-04-02 10-55-04](https://user-images.githubusercontent.com/3522333/55385459-e67ba000-5535-11e9-9b62-7dcf87753963.png)

The same bug also occurs with a simple `/* // */` (the rest of the file is highlighted as a comment):
![Screenshot from 2019-04-02 11-00-19](https://user-images.githubusercontent.com/3522333/55385762-8cc7a580-5536-11e9-8fa0-36b0e5b4c8c2.png)

This is a relatively recent change, I remember the highlighting working correctly about a week ago (but of course, that may depend on when I updated this package).

**Comment by kankaristo** [comment]
Temporary fix is to install this package, which provides Dart syntax highlighting only (works together with this package, but doesn't have the same bug):
https://marketplace.visualstudio.com/items?itemName=oscarcs.dart-syntax-highlighting-only

**Comment by DanTup** [comment]
Yikes! Thanks for the report. This is very likely related to the issue you linked (which shipped on Friday). I'll do some digging.

FWIW, VS Code has shortcuts for quickly commenting out blocks of code that might be easier than this (though it doesn't change that this needs fixing).

**Comment by kankaristo** [comment]
> FWIW, VS Code has shortcuts for quickly commenting out blocks of code that might be easier than this (though it doesn't change that this needs fixing).

I know, but this is "universal" (works in any editor, lots of different languages, etc.), and also nice because it allows you to "toggle" between 2 different code blocks. :)

And I also consider this to be easier/faster, since it's just one character to comment/uncomment the block, instead of highlighting the code block and pressing a keyboard shortcut. At least it's easier once you have the "scaffolding comments" around the code block, before that, maybe not.

But like you said, this should be fixed regardless, since even a simple `/* // */` also breaks highlighting.

**Comment by DanTup** [comment]
Yeah, that makes sense if it's always the same block (though the selection can handle many).

Here's the impact of the change I just made. I've saved this file in the repo now to make it easier to evaluate these in future (it's a bit of a game of cat and mouse - I haven't found an official syntax for the language that is maintained with the changes).

If there's anything other combinations of things you think worth adding, do let me know or (feel free to edit the file in a PR).

![Screenshot 2019-04-02 at 11 06 37 am](https://user-images.githubusercontent.com/1078012/55394813-15931100-5538-11e9-8172-9a0c210d2284.png)

@devoncarew FYI - one of my previous changes to support nested comments resulted in allowing line comments inside block comments, which resulted in the end-of-block-comment marker being ignored if it was on the same line.
